ws.c

```diff
@@ -221,34 +226,54 @@ get_ws_frame(unsigned char *in_buffer, int buf_len, unsigned char **payload_ptr,
 	if (length_field <= 125) {
 		payload_len = length_field;
 	} else if (length_field == 126) { /* msglen is 16bit */
+		uint16_t tmp16;
 		if (buf_len < 4)
 			return INCOMPLETE_DATA;
-		payload_len = ntohs(*(uint16_t *)(in_buffer + 2));
+		memcpy(&tmp16, in_buffer + pos, 2);
+		payload_len = ntohs(tmp16);
 		pos += 2;
 	} else if (length_field == 127) { /* msglen is 64bit */
+		int i;
+		uint64_t tmp64 = 0;
 		if (buf_len < 10)
 			return INCOMPLETE_DATA;
-		payload_len = ntohs(*(uint64_t *)(in_buffer + 2));
-		pos += 8;
+		/* swab bytes from big endian to host byte order */
```

Comment:
```suggestion
		/* swap bytes from big endian to host byte order */
```

Reply:
The term swab is a real term used in programming and it usually refers to endian-swaps or other byte swapping. I'll change it though as per your request.  https://www.man7.org/linux/man-pages/man3/swab.3.html

Reply:
Oh, sorry, haven't heard it before. Thank you for the link. And it is OK to use `swab` here of course.